grpclb/src/main/java/io/grpc/grpclb/GrpclbClientLoadRecorder.java

```diff
@@ -36,44 +35,55 @@
  */
 @ThreadSafe
 final class GrpclbClientLoadRecorder extends ClientStreamTracer.Factory {
+
+  private static final AtomicLongFieldUpdater<GrpclbClientLoadRecorder> callsStartedUpdater =
+      AtomicLongFieldUpdater.newUpdater(GrpclbClientLoadRecorder.class, "callsStarted");
+  private static final AtomicLongFieldUpdater<GrpclbClientLoadRecorder> callsFinishedUpdater =
+      AtomicLongFieldUpdater.newUpdater(GrpclbClientLoadRecorder.class, "callsFinished");
+  private static final AtomicLongFieldUpdater<GrpclbClientLoadRecorder> callsFailedToSendUpdater =
+      AtomicLongFieldUpdater.newUpdater(GrpclbClientLoadRecorder.class, "callsFailedToSend");
+  private static final AtomicLongFieldUpdater<GrpclbClientLoadRecorder>
+      callsFinishedKnownReceivedUpdater =
+          AtomicLongFieldUpdater.newUpdater(
+              GrpclbClientLoadRecorder.class, "callsFinishedKnownReceived");
+
   private final TimeProvider time;
-  private final AtomicLong callsStarted = new AtomicLong();
-  private final AtomicLong callsFinished = new AtomicLong();
+  @SuppressWarnings("unused")
+  private volatile long callsStarted;
+  @SuppressWarnings("unused")
+  private volatile long callsFinished;
 
   // Specific finish types
-  // Access to it should be protected by lock.  Contention is not an issue for these counts, because
-  // normally only a small portion of all RPCs are dropped.
-  @GuardedBy("this")
-  private HashMap<String, AtomicLong> callsDroppedPerToken = new HashMap<String, AtomicLong>();
-  private final AtomicLong callsFailedToSend = new AtomicLong();
-  private final AtomicLong callsFinishedKnownReceived = new AtomicLong();
+  private final ConcurrentMap<String, Long> callsDroppedPerToken =
```

Comment:
Why this change?

I'm not excited to see a change like this among all the noise, without it being called out.

Reply:
Avoids locks, and avoid the extra atomics.  After talking with @zhangkun83 the reason to not use it is because Longs are less dense than AtomicLongs.

Reply:
Ah, I missed the fact it swapped from AtomicLong to Long. I mainly saw the @GuardedBy removal and the ConcurrentMap API usages.